working/augmentation-libraries/parts_with_imports.md

```diff
@@ -0,0 +1,544 @@
+        deferred prefix import scope cannot be extended, and cannot extend
+        another prefix scope, deferred prefix scopes are always linked to a
+        single import directive._
+    *   _It’s possible to look further up in the import chain *C* for a prefix
+        scope to extend. Here it’s chosen that that importing parent file gets
+        to decide which names the part file has access to. If it wants to make
+        a transitive parent import prefix available, it should just not shadow
+        it._
+
+That is: The combined import scope of a Dart file is a chain of the combined
+import scopes of the file and its parent files, each step adding two scopes:
+The (unnamed, top-level) import scope of the unprefixed imports and the prefix
+scope with prefixed imports, each shadowing names further up in the chain.
+
+The *top-level scope* of a Dart file is a library *declaration scope*
+containing every top-level library member declaration in every library or part
+file of the library. The parent scope of the top-level scope of a Dart file is
+the combined import scope of that Dart file. _Each Dart file has its own copy
+of the library declaration scope, all containing the same declarations, because
+the declaration scopes of different files have different parent scopes._
+
+**It’s a compile-time error ** if any file declares an import prefix with the
+same base name as a top-level declaration of the library.
+
+_We have split the prefixes out of the top-level scope, but we maintain that
+they must not have the same names anyway. Any prefix that has the same name as
+a top-level declaration of the library is impossible to reference, because the
+library declaration scope always precedes the prefix scope in any scope chain
+lookup. This does mean that adding a top-level declaration in one part file may
+conflict with a prefix name in another part file in a completely different
+branch of the library file tree. That is not a conflict with the “other file’s
+imports cannot break your code” principle, rather the error is in the file
+declaring the prefix. Other files’ top-level declarations can totally break
+your code. Top-level declarations are global and the unit of ownership is the
+library, so the library author should fix the conflict by renaming the prefix.
+That such a name conflict is a compile-time error, makes it much easier to
+detect if it happens._
+
+### Export directives
+
+Any Dart file can contain an `export` directive. It makes no difference which
+file an `export` is in, its declarations (filtered by any `hide` or `show`
+modifiers) are added to the library’s single export scope, along with those of
+any other  `export`s in the library and the non-private declarations of the
+library itself. Conflicts are handled as usual (as an error if it’s not the
+*same* declaration).
+
+Allowing a part file to have its own export is mainly intended for macro
```

Comment:
Fwiw, macro generated parts can't add exports

Reply:
Well then, it's mainly allowed for consistency then. :)